### ISSUE TITLE
Ability to disable enc checking when not present

### ISSUE THREAD

**Issue by AcuarioCat** [khoih-prog/AsyncWebServer_ESP32_ENC#3]
This is an excellent move forward to provide support for the enc28j60.
Here is the scenario:
A device has 2 variants; one with only WiFi and the other with WiFi and LAN.

A circuit can operate with either LAN or WiFi or both. When the enc is physically present this works perfectly.
The WiFi can be disabled easily so it just works with LAN. This is ok too.
Problem is when the ENC is NOT physically present. 
The return (as expected) from ETH.begin is false and with that it's possible to skip the waitForConnect but the enc continues to be checked/polled/whatever..
The results I see are:

```
E (4742) enc28j60: enc28j60_verify_id(699): wrong chip ID
E (4742) enc28j60: emac_enc28j60_init(1262): verify chip ID failed
E (4747) esp_eth: esp_eth_driver_install(214): init mac failed
esp_eth_driver_install failed
```
This is good and expected and using the false return I skip waitForConnect but then..

```
E (5741) spi_master: check_trans_valid(689): txdata transfer > host maximum
E (5741) enc28j60: enc28j60_do_memory_read(432): spi transmit failed
E (5743) enc28j60: enc28j60_read_packet(572): read memory failed
E (5749) enc28j60: emac_enc28j60_receive(1209): read packet content failed
E (6755) spi_master: check_trans_valid(689): txdata transfer > host maximum
E (6755) enc28j60: enc28j60_do_memory_read(432): spi transmit failed
E (6757) enc28j60: enc28j60_read_packet(572): read memory failed
E (6762) enc28j60: emac_enc28j60_receive(1209): read packet content failed
```

..which is not - and shouldn't happen as the enc is physically missing.

Can this be disabled/fixed..?

**Comment by khoih-prog** [comment]
Hi @AcuarioCat 

Good to know that you can somehow use the library and see it useful.

It's easy to modify the demo-only examples to adapt them to your use-case, as you have seen and tried.

I'm afraid that there is something wrong in your code or core version / installation, etc. as I tried to modify the example code to

```cpp
  bool ETH_ready = ETH.begin( MISO_GPIO, MOSI_GPIO, SCK_GPIO, CS_GPIO, INT_GPIO, SPI_CLOCK_MHZ, SPI_HOST, mac[index] );
  
  if (!ETH_ready)
  {
    Serial.println(F("Not working ENC28J60. Do something else"));
    
    //while(true)
    //  delay(1000);
  }
  else
  {
    // Static IP, leave without this line to get IP via DHCP
    //bool config(IPAddress local_ip, IPAddress gateway, IPAddress subnet, IPAddress dns1 = 0, IPAddress dns2 = 0);
    ETH.config(myIP, myGW, mySN, myDNS);
  
    ESP32_ENC_waitForConnect();
  }
```

from 

```cpp
  ETH.begin( MISO_GPIO, MOSI_GPIO, SCK_GPIO, CS_GPIO, INT_GPIO, SPI_CLOCK_MHZ, ETH_SPI_HOST, mac[index] );


  // Static IP, leave without this line to get IP via DHCP
  //bool config(IPAddress local_ip, IPAddress gateway, IPAddress subnet, IPAddress dns1 = 0, IPAddress dns2 = 0);
  //ETH.config(myIP, myGW, mySN, myDNS);


  ESP32_ENC_waitForConnect();
```

and it's still OK to test on bare ESP32 (no ENC28J60 at all)


```
Start AsyncSimpleServer_ESP32_ENC on ESP32_DEV with ESP32_ENC28J60
AsyncWebServer_ESP32_ENC v1.6.3 for core v2.0.0+
[AWS] Default SPI pinout:
[AWS] MOSI: 23
[AWS] MISO: 19
[AWS] SCK: 18
[AWS] CS: 5
[AWS] INT: 4
[AWS] SPI Clock (MHz): 8
[AWS] =========================
E (262) enc28j60: enc28j60_verify_id(699): wrong chip ID
E (262) enc28j60: emac_enc28j60_init(1262): verify chip ID failed
E (262) esp_eth: esp_eth_driver_install(214): init mac failed
esp_eth_driver_install failed
Not working ENC28J60. Do something else
HTTP EthernetWebServer is @ IP : 0.0.0.0
```


You certainly have to write **correct** code to handle the missing ENC28J60 case, such as switching to use WiFi, etc

**Comment by AcuarioCat** [comment]
@khoih-prog 
Thanks for the reply - this has driven me crazy but now I know how you can reproduce it.
I tried on a dev board and, as you say, it works fine but..
I have an SD card as well.
The SD card is connected to the SPI bus and has it's own CS.
With an SD card physically connected you get the messages I showed above. If it is not connected then you don't.

SD card is connected as follows:
MOSI:23
MISO:19
SCK:18
CS:5

The SD card holder is a standard AliExpress Micro SD card adaptor.

..even easier, if you don't have a SD card adaptor, just pull MISO up to 3V3 with a 10K resistor and the messages appear.

**Comment by AcuarioCat** [comment]
After some testing and debugging I have found that the driver is not delete from the task list if it fails to initialize (which it won't do if it's missing).
I have modified esp_eth_mac_enc28j60.c as follows:

- Add a forward declaration to : static esp_err_t emac_enc28j60_del(esp_eth_mac_t* mac); (reordering the code would also work).
- add a call to emac_enc28j60_del(mac); in the emac_enc28j60_init routine just before the return.

With these changes when the driver fails its task is no longer run so no error messages.
